ansible/playbooks/deploy-cinder-netapp-volumes-reference.yaml

```diff
@@ -0,0 +1,216 @@
+        dest: "{{ item.dest }}"
+        owner: root
+        group: root
+        mode: "{{ item.mode | default('0644') }}"
+      loop:
+        - src: "{{ playbook_dir }}/templates/cinder-netapp-volume.service"
+          dest: /etc/systemd/system/cinder-netapp-volume.service
+        - src: "{{ playbook_dir }}/templates/sudoers"
+          dest: /etc/sudoers.d/cinder-volume
+          mode: "0440"
+      notify:
+        - Restart cinder-volume-netapp systemd services
+
+    - name: Create the cinder-volume-netapp service configuration
+      ansible.builtin.copy:
+        content: "{{ _kubernetes_cinder_etc_secret.resources[0].data[item.src] | b64decode }}"
+        dest: "{{ item.dest }}"
+        owner: root
+        group: root
+        mode: "0644"
+      loop:
+        - src: "logging.conf"
+          dest: /etc/cinder/logging.conf
+        - src: "volume.filters"
+          dest: /etc/cinder/volume.filters
+      notify:
+        - Restart cinder-volume-netapp systemd services
+
+    - name: Create the cinder-volume-netapp configuration stage file
+      changed_when: false
+      ansible.builtin.copy:
+        content: "{{ _kubernetes_cinder_etc_secret.resources[0].data['cinder.conf'] | b64decode }}"
+        dest: "/etc/cinder/cinder.conf.stage"
+        owner: root
+        group: root
+        mode: "0644"
+
+    - name: Replace the host in the cinder.conf.stage with the current Ansible FQDN in the stage file
+      changed_when: false
+      community.general.ini_file:
+        path: "/etc/cinder/cinder.conf.stage"
+        section: DEFAULT
+        option: host
+        value: "{{ ansible_fqdn }}"
+        create: true
+
+    - name: Ensure the backend configuration is set to our expected value
+      changed_when: false
+      community.general.ini_file:
+        path: "/etc/cinder/cinder.conf.stage"
+        section: DEFAULT
+        option: enabled_backends
+        value: "{{ cinder_backend_name }}"
+        create: true
+
+    - name: Create the cinder-volume-netapp configuration
+      ansible.builtin.copy:
+        src: "/etc/cinder/cinder.conf.stage"
+        dest: "/etc/cinder/cinder.conf"
+        owner: root
+        group: root
+        mode: "0644"
+        remote_src: true
+      notify:
+        - Restart cinder-volume-netapp systemd services
+
+    - name: Set enabled backend fact
+      set_fact:
+        cinder_backend: |
+          {% set rendered_backend = {} %}
+          {% for backend in cinder_backend_name.split(',') %}
+          {%   set default_backend = (_kubernetes_cinder_etc_secret.resources[0].data['backends.conf'] | b64decode | community.general.from_ini)[backend] %}
+          {%   set network_interface = hostvars[inventory_hostname][cinder_storage_network_interface]['ipv4']['address'] | default(ansible_default_ipv4.address) %}
+          {%   set network_interface_secondary = hostvars[inventory_hostname][cinder_storage_network_interface_secondary]['ipv4']['address'] %}
```

Comment:
Not sure if this should be fully wrapped in the multipath conditional below, but need to test this playbook on one node to see how the files lay down. Overall, this is an excellent first pass.